test/lib/validation/ValidationError.js

```diff
@@ -520,6 +520,163 @@ property2:
 		"analyzeYamlError should return expected results");
 });
 
+test.serial("ValidationError.analyzeYamlError: documentIndex=0 (Without leading separator)", (t) => {
+	const error = {dataPath: "/property3"};
+	const yaml = {
+		path: "/my-project/ui5.yaml",
+		source:
+`property1: value1document1
+property2: value2document1
+property3: value3document1
+property4: value4document1
+property5: value5document1
+---
+property1: value1document2
+property2: value2document2
+property3: value3document2
+property4: value4document2
+property5: value5document2`,
+		documentIndex: 0
+	};
+
+	const info = ValidationError.analyzeYamlError({error, yaml});
+
+	t.deepEqual(info, {line: 3, column: 1},
+		"analyzeYamlError should return expected results");
+});
+
+test.serial("ValidationError.analyzeYamlError: documentIndex=0 (With leading separator)", (t) => {
+	const error = {dataPath: "/property3"};
+	const yaml = {
+		path: "/my-project/ui5.yaml",
+		source:
+`---
```

Comment:
What about one (or multiple) newlines at the beginning of the file?

Reply:
Good point. Done 👍